src/lib/storage/table.cpp

```diff
@@ -151,19 +159,32 @@ uint32_t Table::max_chunk_size() const { return _max_chunk_size; }
 
 std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) {
   DebugAssert(chunk_id < _chunks.size(), "ChunkID " + std::to_string(chunk_id) + " out of range");
-  return _chunks[chunk_id];
+  if (_type == TableType::References) {
+    // Locking for reference tables is wasted performance since they are not accessed by multiple concurring threads
+    return _chunks[chunk_id];
+  } else {
+    return std::atomic_load(&_chunks[chunk_id]);
+  }
 }
 
 std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
   DebugAssert(chunk_id < _chunks.size(), "ChunkID " + std::to_string(chunk_id) + " out of range");
-  return _chunks[chunk_id];
+  if (_type == TableType::References) {
```

Comment:
Please add a comment to the declaration of `_chunks` on why (1) we need to access it atomically and (2) why we cannot use `atomic<shared_ptr>`